codex-rs/core/tests/suite/rmcp_client.rs

```diff
@@ -369,3 +559,60 @@ async fn wait_for_streamable_http_server(
         sleep(Duration::from_millis(50)).await;
     }
 }
+
+fn write_fallback_oauth_tokens(
+    home: &Path,
+    server_name: &str,
+    server_url: &str,
+    client_id: &str,
+    access_token: &str,
+    refresh_token: &str,
+) -> anyhow::Result<()> {
+    let expires_at = SystemTime::now()
+        .checked_add(Duration::from_secs(3600))
+        .ok_or_else(|| anyhow::anyhow!("failed to compute expiry time"))?
+        .duration_since(UNIX_EPOCH)?
+        .as_millis() as u64;
+
+    let store = serde_json::json!({
+        "stub": {
+            "server_name": server_name,
+            "server_url": server_url,
+            "client_id": client_id,
+            "access_token": access_token,
+            "expires_at": expires_at,
+            "refresh_token": refresh_token,
+            "scopes": ["profile"],
+        }
+    });
+
+    let file_path = home.join(".credentials.json");
+    fs::write(&file_path, serde_json::to_vec(&store)?)?;
+    Ok(())
+}
+
+struct EnvVarGuard {
```

Comment:
Hmm, I see you added `#[serial(codex_home)]`, but this still concerns me because I know people will be tempted to copy/paste this utility and will inevitably introduce flaky, hard-to-debug tests.

Reply:
Chatted about this offline